### ISSUE TITLE
[4.0] Backend - Mark Search Tools region as landmark

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#20753]
**Accessibility problem:** 
The key area on the pages of the "manager" type is the search tools. However, they are not marked as a landmark in the page code. The toolkit has no label or heading. Screen reader users cannot quickly access the main workspace by navigating through landmarks. A person who uses only the keyboard also has difficulty accessing these tools.
**Step to reproduce issue:**
1. Go to Article manager page of the Joomla 4 Backend.
2. **Inspect code**. Use the developer's tools in your browser to explore the code. 
3. **Keyboard access**. Try to access the search tools only using the keyboard (use Tab key).
4. **Screen reader access**. Use the screen reader to navigate through the page.
* NVDA: 
-- use D key to navigate through landmarks
-- use NVDA key+F7 to open a list of links, headings, form fields, buttons and landmarks for a user to navigate (NVDA key = Insert or CapsLock)
* JAWS
-- use R key to navigate through landmarks
-- use Control+Insert+R to open a list of all landmarks  for a user to navigate
* VoiceOver (desktop)
-- use Ctrl+Option+U to open Rotor, next use left or right arrows to navigate to the Landmarks menu. 

**Expected result**: You have access to a variety of landmarks, including search landmark
**Current result**: If you are using a screen reader, you can only access two landmarks - banner and navigation, but no search  landmark. If you only use the keyboard, you can access it, but you must press the Tab key several times (tools are accessible but difficult to use).

### Success criteria:
* [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG20/quickref/#qr-content-structure-separation-programmatic): Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text. (Level A)
* [2.4.6 Headings and Labels](https://www.w3.org/WAI/WCAG20/quickref/#qr-navigation-mechanisms-descriptive): Headings and labels describe topic or purpose. (Level AA)
* [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/quickref/#name-role-value): For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies. (Level A)

### Techniques:
[ARIA11](https://www.w3.org/TR/WCAG20-TECHS/ARIA11): Using ARIA landmarks to identify regions of a page
[H69](https://www.w3.org/TR/WCAG20-TECHS/H69): Providing heading elements at the beginning of each section of content
[G141](https://www.w3.org/TR/WCAG20-TECHS/G141): Organizing a page using headings
[H42](https://www.w3.org/TR/WCAG20-TECHS/H42): Using h1-h6 to identify headings
[H69](https://www.w3.org/TR/WCAG20-TECHS/H69): Providing heading elements at the beginning of each section of content
### WAI-ARIA practices 
[4.3.8. Search](https://www.w3.org/TR/wai-aria-practices-1.1/#aria_lh_search): A search landmark contains a collection of items and objects that, as a whole, combine to create search functionality.
* Use the search landmark instead of the form landmark when the form is used for search functionality.
* If a page includes more than one search landmark, each should have a unique label...
### Tutorials, articles:
* [Web Accessibility Tutorials: Page structure tutorial](https://www.w3.org/WAI/tutorials/page-structure/)
* [ARIA landmarks example](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/index.html)
* [Accessible landmarks by Scot O’Hara](https://www.scottohara.me/blog/2018/03/03/landmarks.html)
## Suggested solution
* Correct the code of the search tools (path: layouts/joomla/searchtools). Use the role="search" attribute to indicate a landmark. 
* Add a heading or label  'Search Tools' to screen reader only. It will be used in a special menu and by a screen reader.

**Comment by brianteeman** [comment]
> Correct the code of the search tools (path: layouts/joomla/searchtools). Use the role="search" attribute to indicate a landmark.

Pull request incoming

> Add a heading or label 'Search Tools' to screen reader only. It will be used in a special menu and by a screen reader.

Not needed as the landmark will be recognised by screen readers

**Comment by zwiastunsw** [comment]
> Not needed as the landmark will be recognised by screen readers

I agree. But if you want to create a special menu, you can use a label or a heading in it. Screen reader users can also navigate through the headings.

**Comment by brianteeman** [comment]
Any "special menu" should be able to read a landmark role by default but anyway there is already a label present

I have done a PR to add the search role #20754 
And another PR to remove the incorrect toolbar role on the searchbox #20755

**Comment by brianteeman** [comment]
All done and merged
